ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterMapUtils.java

```diff
@@ -166,5 +168,43 @@ static void validateDiskCapacity(long diskCapacityInBytes) {
           "Invalid disk capacity: " + diskCapacityInBytes + " is more than " + MAX_DISK_CAPACITY_IN_BYTES);
     }
   }
+
+  /**
+   * Gets a map from datacenter name to datacenter id. The id generated for each datacenter name is unique and
+   * non-negative.
+   * @param dcNameSet A set of datacenter names.
+   * @return A map from datacenter name to datacenter id in short.
+   * @throws IllegalStateException if the generated id of some datacenter collide.
+   */
+  static Map<String, Short> getDcNameToIdMap(Set<String> dcNameSet) {
+    if (dcNameSet == null) {
+      throw new IllegalArgumentException("dcNames cannot be null.");
+    }
+    Map<String, Short> datacenterToIdMap = new HashMap<>();
+    Set<Short> dcIdSet = new HashSet<>();
+    for (String dcName : dcNameSet) {
+      short dcId = hashDcNameToNonNegativeId(dcName);
```

Comment:
I feel that all this hashing is an overkill. Having the clustermap statically set a unique id for every datacenter seems simpler.